### ISSUE TITLE
Not connecting in ubuntu 20.04

### ISSUE THREAD

**Issue by the-loudspeaker** [magne4000/quassel-webserver#285]
Used to work fine in Ubuntu 18.04 but does not work in 20.04. The client connects and disconnects immediately.
There seems to be some tls error. Here are the error logs from quasselcore: https://p.ip.fi/HYvY

**Comment by digitalcircuit** [comment]
As a workaround, when connecting, expand `more options` and disable `SSL core connection`.  Not ideal, but if your Quassel core is `localhost`, it's not the end of the world.

My current guess is there's an issue with [`node-libquassel`](https://github.com/magne4000/node-libquassel ) with newer (TLSv1.3) setups.  Instead of `secureProtocol` in [`createSecureContext() options`](https://nodejs.org/docs/latest-v12.x/api/tls.html#tls_tls_createsecurecontext_options ), `libquassel` should use `minVersion`… but I'm still figuring out how to test these changes.

**Comment by the-loudspeaker** [comment]
> As a workaround, when connecting, expand `more options` and disable `SSL core connection`.  Not ideal, but if your Quassel core is `localhost`, it's not the end of the world.
> 
> My current guess is there's an issue with [`node-libquassel`](https://github.com/magne4000/node-libquassel ) with newer (TLSv1.3) setups.  Instead of `secureProtocol` in [`createSecureContext() options`](https://nodejs.org/docs/latest-v12.x/api/tls.html#tls_tls_createsecurecontext_options ), `libquassel` should use `minVersion`… but I'm still figuring out how to test these changes.
> 
> **Edit:** After other edit attempts failed, manually editing the compiled `libquassel.js` doesn't seem to work, either - setting `minVersion` still fails.
> 
> **Edit 2:** I've not been able to figure out the SSL/TLS bug, so I've set up `securecore: false` by default on Ubuntu 20.04+ since it's `localhost` anyways.  Again, not ideal, but ah well.

Yes i had it localhost and it worked. But now my quasselcore shifted to another machine so now I cannot use this workaround. Seems it is referenced in a commit. Will check that out and see how it works.

**Comment by the-loudspeaker** [comment]
I have just tried it in ubuntu 20.04, my quasselcore is in ubuntu 18.04. Doesn't directly work with ssl core connection enabled in quassel-webserver. I have to disable it once, connect to my core. then logout and then from next time onwards it works with ssl core connection ticked. 

after that it works like normal.

**Comment by magne4000** [comment]
This comment can be of interest https://github.com/magne4000/node-libquassel/issues/25#issuecomment-988140501

**Comment by mtrolley** [comment]
The suggestion above about disabling "SSL core connection" doesn't work for me. The core log still shows even with that option unchecked:
```
2023-08-02 18:11:42 [Info ] Client connected from 172.18.0.6
2023-08-02 18:11:42 [Warn ] "Socket error 13: Error during SSL handshake: error:0A000076:SSL routines::no suitable signature algorithm"
2023-08-02 18:11:42 [Info ] Non-authed client disconnected: 172.18.0.6
```
